wasm/jit/jit_amd64.go

```diff
@@ -1096,14 +1103,103 @@ func (b *amd64Builder) handleLe(o *wazeroir.OperationLe) error {
 	return nil
 }
 
+func (b *amd64Builder) handleConstI32(o *wazeroir.OperationConstI32) error {
+	reg, err := b.allocateRegister(gpTypeInt)
+	if err != nil {
+		return err
+	}
+	loc := b.locationStack.pushValueOnRegister(reg)
+	loc.setValueType(wazeroir.SignLessTypeI32)
+
+	prog := b.newProg()
+	prog.As = x86.AMOVL // Note 32-bit move!
+	prog.From.Type = obj.TYPE_CONST
+	prog.From.Offset = int64(o.Value)
+	prog.To.Type = obj.TYPE_REG
+	prog.To.Reg = reg
+	b.addInstruction(prog)
+	return nil
+}
+
 func (b *amd64Builder) handleConstI64(o *wazeroir.OperationConstI64) error {
 	reg, err := b.allocateRegister(gpTypeInt)
 	if err != nil {
 		return err
 	}
 	loc := b.locationStack.pushValueOnRegister(reg)
 	loc.setValueType(wazeroir.SignLessTypeI64)
-	b.movConstToRegister(int64(o.Value), reg)
+
+	prog := b.newProg()
+	prog.As = x86.AMOVQ
+	prog.From.Type = obj.TYPE_CONST
+	prog.From.Offset = int64(o.Value)
+	prog.To.Type = obj.TYPE_REG
+	prog.To.Reg = reg
+	b.addInstruction(prog)
+	return nil
+}
+
+func (b *amd64Builder) handleConstF32(o *wazeroir.OperationConstF32) error {
+	reg, err := b.allocateRegister(gpTypeFloat)
+	if err != nil {
+		return err
+	}
+	loc := b.locationStack.pushValueOnRegister(reg)
+	loc.setValueType(wazeroir.SignLessTypeF32)
+
+	// We cannot directly load the value from memory to float regs,
+	// so we move it to int reg temporarily.
+	tmpReg, err := b.allocateRegister(gpTypeInt)
+	if err != nil {
+		return err
+	}
+	moveToTmpReg := b.newProg()
+	moveToTmpReg.As = x86.AMOVL // Note 32-bit mov!
+	moveToTmpReg.From.Type = obj.TYPE_CONST
+	moveToTmpReg.From.Offset = int64(uint64(math.Float32bits(o.Value)))
+	moveToTmpReg.To.Type = obj.TYPE_REG
+	moveToTmpReg.To.Reg = tmpReg
+	b.addInstruction(moveToTmpReg)
+
+	prog := b.newProg()
+	prog.As = x86.AMOVL // Note 32-bit mov!
+	prog.From.Type = obj.TYPE_REG
+	prog.From.Reg = tmpReg
```

Comment:
do we need to release this?

Reply:
https://github.com/tetratelabs/wazero/pull/82/commits/36f609505d3ff189a3a85a44cce5a1c95c02b13e